### ISSUE TITLE
Thalia Pay sales dialog opens twice,

### ISSUE THREAD

**Issue by DeD1rk** [svthalia/Reaxit#455]
### Describe the bug
I've heard, but not yet seen, that when you pay with thalia pay for a sales order, the dialog stays open after paying. Might not be accurate but worth trying out.

### How to reproduce
Pray to the demo-gods.

**Comment by JAicewizard** [comment]
Is there any way we can try to reproduce this next meeting? You're more fimiliar with thaliapay so we'll probably need to figure it out together

**Comment by DeD1rk** [comment]
We can just give it a try on staging

**Comment by DeD1rk** [comment]
Almost forgot, but I reproduced on the borrel. It felt to me like the problem was that the sales dialog stays open until the payment succeeds, but the confirmation dialog closes immediately when you push the button. So as the server is apparently quite slow, that allows you to press 'pay' multiple times.

Here's a video. I started it a bit late but clear enough. https://github.com/svthalia/Reaxit/assets/41264528/17596c8d-727a-4605-a583-85c99205c5c4

Also, I think before the 'pay' request can be sent, the 'claim order' request must also have finished. So probably we also need to temporarily disable the pay button until that's done. Possibly the first few paying attempts fail because of the order not having been claimed yet?
